src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDeliveryService.cs

```diff
@@ -535,6 +535,8 @@ public class PurchaseOrderDeliveryService : IPurchaseOrderDeliveryService
 
             detail.PurchaseDeliveries = newDeliveries.ToList();
 
+            this.purchaseOrderService.UpdateOrder(order, order, privileges);
```

Comment:
It's the only really Public way through to that part of the service. It definitely wasn't designed for this. Is there any other way I should be getting to it? I could make some other functions in that service Public but didn't think that was a great idea. I do get that this isn't the most readable thing i nthe world though.

Reply:
This is due to making use of the PurchaseOrderService and making sure that the detail calculations are being performed there rather than writing the same function as well a few other parts of that service in the deliveries. That could be an alternative or I could open up a more direct route?

Reply:
ooh I see what you mean. If it works it works - A lot of this will probably end up getting refactored anyways